### PR TITLE
[do not merge] Debug only: debug failing tests on emulator

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [ ] Tests written, or not not needed

### CODE REVIEW

**Comment by hannesa2** [comment]
It needs a rebase.

Independend of this you use probable the wrong Espresso action, with https://github.com/hannesa2/LiveEdgeDetection/blob/master/.github/workflows/Android-CI-Espresso.yml#L45-L50

you can do this

      - name: Save logcat output
        uses: actions/upload-artifact@master
        if: failure()
        with:
          name: Nextcloud-logcat
          path: artifacts/logcat.log

This would be one of my next task, but I've the fear to change **ALL** screenshots.

With your approach, I guess you see a mixture of any outputs. Gradle and logcat

**Comment by tobiasKaminsky** [comment]
This is for now a test PR, to get more debug output.

One finding yesterday was, that for some reasons server on drone just dropped out and thus the tests failed, as they require a server connection.

File: .github/workflows/screenShotTest.yml [hunk 1]
```diff
@@ -35,11 +35,16 @@ jobs:
                     emulator-options: -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim -skin 500x833
                     script: ./gradlew gplayDebugExecuteScreenshotTests -Pandroid.testInstrumentationRunnerArguments.annotation=com.owncloud.android.utils.ScreenshotTest -Pandroid.testInstrumentationRunnerArguments.COLOR=${{ matrix.color }} -Pandroid.testInstrumentationRunnerArguments.DARKMODE=${{ matrix.scheme }}
             -   name: upload failing results
+                uses: actions/upload-artifact@master
                 if: ${{ failure() }}
+                with:
+                    name: Nextcloud-logcat
+                    path: artifacts/logcat.log
```
Comment (hannesa2) [inline .github/workflows/screenShotTest.yml:42]: I'm not sure if `uses: reactivecircus/android-emulator-runner@v2` produces this logcat file.
I know `uses: malinskiy/action-android/install-sdk@release/0.1.0` produces a logcat file. 
But I've seen only misleading entries. It needs maybe more investigation

File: .drone.yml [hunk 2]
```diff
@@ -109,6 +111,8 @@ steps:
       - ./gradlew jacocoTestGplayDebugUnitTestReport && scripts/deleteOutdatedComments.sh "master" "Unit" $DRONE_PULL_REQUEST $GIT_USERNAME $GIT_TOKEN || scripts/uploadReport.sh $LOG_USERNAME $LOG_PASSWORD $DRONE_BUILD_NUMBER "master" "Unit" $DRONE_PULL_REQUEST $GIT_USERNAME $GIT_TOKEN
       - ./gradlew installGplayDebugAndroidTest
       - scripts/wait_for_server.sh "server"
+      - adb logcat -c
+      - adb logcat &
```
Comment (hannesa2) [inline .drone.yml:115]: I guess this is the right way, to collect output

`adb logcat > artifacts/logcat.log &` only file

or

`adb logcat | tee artifacts/logcat.log &` file and stdout

**Comment by tobiasKaminsky** [comment]
> stable-IT test failed: https://www.kaminsky.me/nc-dev/android-integrationTests/838-IT-stable-14-12

Here I see:
> Test failed to run to completion. Reason: 'Instrumentation run failed due to 'Process crashed.''. Check device logcat for details

Downloading and analyzing: https://drone.nextcloud.com/nextcloud/android/838/1/3 

```
02-24 14:12:47.016  4726  4813 D EncryptionTestIT: 4734 of 5000
--------- beginning of crash
02-24 14:12:47.017  4726  6770 E AndroidRuntime: FATAL EXCEPTION: Thread-104
02-24 14:12:47.017  4726  6770 E AndroidRuntime: Process: com.******.client, PID: 4726
02-24 14:12:47.017  4726  6770 E AndroidRuntime: java.lang.RuntimeException: java.net.SocketTimeoutException: failed to connect to server.com/52.8.126.80 (port 443) from /192.168.232.2 (port 47107) after 10000ms
02-24 14:12:47.017  4726  6770 E AndroidRuntime: 	at com.******.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:174)
02-24 14:12:47.017  4726  6770 E AndroidRuntime: 	at com.******.client.network.ConnectivityServiceImpl.isInternetWalled(ConnectivityServiceImpl.java:75)
02-24 14:12:47.017  4726  6770 E AndroidRuntime: 	at com.owncloud.android.ui.helpers.FileOperationsHelper.lambda$startSyncForFileAndIntent$0$FileOperationsHelper(FileOperationsHelper.java:224)
02-24 14:12:47.017  4726  6770 E AndroidRuntime: 	at com.owncloud.android.ui.helpers.-$$Lambda$FileOperationsHelper$gOA6CuCeyRvDYszIXZUBLTgMurE.run(Unknown Source:6)
02-24 14:12:47.017  4726  6770 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:764)
02-24 14:12:47.017  4726  6770 E AndroidRuntime: Caused by: java.net.SocketTimeoutException: failed to connect to server.com/52.8.126.80 (port 443) from /192.168.232.2 (port 47107) after 10000ms
```

Opening corresponding server log from https://drone.nextcloud.com/nextcloud/android/838/1/2 : 
`"reqId":"1vm2yvoNh6L42Xq0VOxv","level":1,"time":"2021-02-24T14:12:34+00:00"`

Last reply from server was ~13s before.
So call from Android did never received an answer as server went away?

**Comment by tobiasKaminsky** [comment]
Another finding:
This seems to crash as it cannot connect to server.com, which is of course true as server.com is not a NC server.
Via drone we use http://server/ to test with real server (fired up via docker).

So it seems that somewhere a real test is run against server.com, which is wrong.

**Comment by AndyScherzinger** [comment]
> So it seems that somewhere a real test is run against server.com, which is wrong.

@tobiasKaminsky we have quite a lot of tests that use accountsa which have server.com as their account information, so maybe that's the issue then, no?

**Comment by AlvaroBrey** [comment]
Superseded by #8791

**Pull request closed without merging**